chatkit-core/src/main/kotlin/com/pusher/chatkit/rooms/RoomStore.kt

```diff
@@ -48,79 +46,119 @@ internal class RoomStore(
             event: UserSubscriptionEvent
     ): List<UserSubscriptionEvent> =
             when (event) {
+                // TODO: extract well named funs with clear responsibility
                 is UserSubscriptionEvent.InitialState -> {
                     val knownRooms = this.toList()
-                    val removedFrom = knownRooms.filterNot {
-                        event.rooms.contains(it)
-                    }.onEach {
-                        this -= it
-                    }.map {
-                        UserSubscriptionEvent.RemovedFromRoomEvent(it.id)
-                    }
 
                     val addedTo = event.rooms.filterNot {
                         knownRooms.contains(it)
                     }.onEach {
                         this += it
+                    }.map { room ->
+                        UserSubscriptionEvent.AddedToRoomEvent(room,
+                                event.readStates.find { it.roomId == room.id }!!,
```

Comment:
Pretty sure this isn't guaranteed to be present :(

Reply:
Good spot, thanks! I'll address that. Definitely should be addressed from the perspecitve of getting subsequent InitialState on reconnect. However when it comes to when the `added-to-room` event is coming from the backend isn't it guaranteed for read state to always be there, i.e. when the current user is added to a room that falls beyond the top 1000? @callum-oakley maybe you could check that and let us know the answer please?

Reply:
Confirmed with @callum-oakley that read state is always there for `added_to_room` event and [fixed](https://github.com/pusher/chatkit-android/pull/139/commits/f7bfebc361b30c024fae67390556b88632e027c6)